### PR TITLE
REST-95 : Add Support for AddSuffixToColumnNames transformation

### PR DESCRIPTION
**Pull request by mandarmarathe123**

### CODE REVIEW

File: restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_add_suffix_to_column_names_transformation.conf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    url = "http://localhost:8080/add-suffix-to-column-names-transformation"
+  }
+
+  data-response = {
+    body-format = "JSON"
+
+    transformations = [
+      {
+        type = "add-suffix-to-column-names"
+        suffix = "_old"
```
Comment (rahulbhatia023) [inline restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_add_suffix_to_column_names_transformation.conf:14]: Here, the undersore should be added by the application itself and not the user.

File: restonomer-core/src/main/scala/com/clairvoyant/restonomer/core/transformation/RestonomerTransformation.scala [hunk 2]
```diff
@@ -82,3 +82,13 @@ case class ReplaceStringInColumnValue(
     restonomerResponseDF.replaceStringInColumnValue(columnName, pattern, replacement)
 
 }
+
+case class AddSuffixToColumnNames(
+    suffix: String,
+    columnNames: List[String] = null
```
Comment (rahulbhatia023) [inline restonomer-core/src/main/scala/com/clairvoyant/restonomer/core/transformation/RestonomerTransformation.scala:88]: Initialise it as empty list and not null

File: restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala [hunk 3]
```diff
@@ -63,6 +63,25 @@ object DataFrameTransformerImplicits {
     def replaceStringInColumnValue(columnName: String, pattern: String, replacement: String): DataFrame =
       df.withColumn(columnName, regexp_replace(col(columnName), pattern, replacement))
 
+    def addSuffixToColNames(suffix: String, columnNames: List[String]): DataFrame = {
+
+      if (columnNames == null || columnNames.isEmpty)
```
Comment (rahulbhatia023) [inline restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala:68]: Remove null condition

File: restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala [hunk 4]
```diff
@@ -63,6 +63,25 @@ object DataFrameTransformerImplicits {
...
+        df.select(
+          df.columns.map(columnName => df(columnName).alias(columnName + suffix)): _*
```
Comment (rahulbhatia023) [inline restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala:70]: add undersore here

**Review by rahulbhatia023 — CHANGES_REQUESTED** [review]
Suggested few changes

File: restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_add_suffix_to_column_names_transformation.conf (hunk 1 above)
Comment (mandarmarathe123) [reply restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_add_suffix_to_column_names_transformation.conf]: Done! Underscore will be added by the application itself now.

File: restonomer-core/src/main/scala/com/clairvoyant/restonomer/core/transformation/RestonomerTransformation.scala (hunk 2 above)
Comment (mandarmarathe123) [reply restonomer-core/src/main/scala/com/clairvoyant/restonomer/core/transformation/RestonomerTransformation.scala]: Done!

File: restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala (hunk 3 above)
Comment (mandarmarathe123) [reply restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala]: Removed!

File: restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala (hunk 4 above)
Comment (mandarmarathe123) [reply restonomer-spark-utils/src/main/scala/com/clairvoyant/restonomer/spark/utils/transformer/DataFrameTransformerImplicits.scala]: Added required underscore character here.

**Pull request merged**